components/speedreader/speedreader_url_loader.cc

```diff
@@ -272,9 +264,11 @@ void SpeedReaderURLLoader::MaybeLaunchSpeedreader() {
                 return data;
               }
 
-              return GetDistilledPageResources() + transformed;
+              return stylesheet + transformed;
             },
-            std::move(buffered_body_), whitelist_->MakeRewriter(response_url_)),
+            std::move(buffered_body_),
+            rewriter_service_->MakeRewriter(response_url_),
```

Comment:
it's not safe to access `rewriter_service_` here because it isn't thread safe and also since this is posted to a thread pool and not a url loader owned task runner, there is no guarantee that the service will still be valid when this runs

Reply:
also all rust code must be called from the same thread per internal guidelines